src/util/animate_color.ts

```diff
@@ -31,49 +24,45 @@ import { Color } from "../color";
    * @param {Function} [options.abort] Additional function with logic. If returns true, onComplete is called.
    * @returns {Function} abort function
    */
-  function animateColor(fromColor, toColor, duration, options) {
-    var startColor = new Color(fromColor).getSource(),
+export function animateColor(fromColor, toColor, duration, options = {}) {
+  const startColor = new Color(fromColor).getSource(),
         endColor = new Color(toColor).getSource(),
         originalOnComplete = options.onComplete,
         originalOnChange = options.onChange;
-    options = options || {};
 
-    return fabric.util.animate(Object.assign(options, {
-      duration: duration || 500,
-      startValue: startColor,
-      endValue: endColor,
-      byValue: endColor,
-      easing: function (currentTime, startValue, byValue, duration) {
-        var posValue = options.colorEasing
-          ? options.colorEasing(currentTime, duration)
-          : 1 - Math.cos(currentTime / duration * (Math.PI / 2));
-        return calculateColor(startValue, byValue, posValue);
-      },
-      // has to take in account for color restoring;
-      onComplete: function(current, valuePerc, timePerc) {
-        if (originalOnComplete) {
-          return originalOnComplete(
-            calculateColor(endColor, endColor, 0),
+  return fabric.util.animate(Object.assign(options, {
```

Comment:
spread object?

Reply:
better of
I think this object should be a class that handles the animation state, constructor accepting options.

Reply:
why? Animations has a lot of options yes, but that does not mean the thosee needs to be a class. if we es6fy it better it i will probably be less complex